mongo/submission.py

```diff
@@ -745,12 +751,18 @@ def own_permission(self, user) -> Permission:
             cap = self.Permission.MANAGER
         elif user.username == self.user.username:
             cap = self.Permission.STUDENT
-        elif Problem(self.problem).permission(user=user,
-                                              req=Problem.Permission.VIEW):
+        elif Problem(self.problem).permission(
+                user=user,
+                req=Problem.Permission.VIEW,
+        ):
             cap = self.Permission.OTHER
         else:
             cap = self.Permission(0)
 
+        CE = 2
+        if self.status == CE and self.user.username == user.username:
+            cap |= self.Permission.VIEW_OUTPUT
```

Comment:
`self.user.username == user.username` seems redundant, we have defined this as `STUDENT` in above.

Also, does it imply that `MANAGER`/teacher cannot view output?

Reply:
> `self.user.username == user.username` seems redundant, we have defined this as `STUDENT` in above.

So you suggest to move CE check to the student block?

> 
> Also, does it imply that `MANAGER`/teacher cannot view output?

No, `MANAGER` can view ouput unconditionally, no extra check needed.

Reply:
1. Nah, I suggest to keep CE here and replace `self.user.username == user.username` with `STUDENT`

2. Got it, thanks